### PR TITLE
feat(role): add flags

### PR DESCRIPTION
**Pull request by Earlopain**
Depends on #1334

https://discord.com/developers/docs/topics/permissions#role-object-role-flags

A better solution to what I was doing in my onboarding pr, wasn't documented at the time though.
I also thought about adding the `tags` field as per #1316 but that has weird nullable types I didn't want to bother with. The issue should also be reopened as the PR doesn't address it fully and isn't merged.

### CODE REVIEW

File: structs.go [hunk 1]
```diff
@@ -1239,13 +1239,51 @@ type Role struct {
...
+	// This is a combination of bit masks; the presence of a certain permission can
+	// be checked by performing a bitwise AND between this int and the flag.
+	Flags RoleFlags `json:"flags"`
 }
 
+// RoleFlags is the flags of "role" (see RoleFlags* consts)
+// https://discord.com/developers/docs/topics/permissions#role-object-role-flags
+type RoleFlags int
+
+// Block containing known RoleFlags values
```
Comment (FedorLap2006) [inline structs.go:1259]:
```suggestion
// Block containing known RoleFlags values.
```

File: structs.go [hunk 2]
```diff
@@ -1239,13 +1239,51 @@ type Role struct {
...
+const (
+	// RoleFlagInPrompt Role can be selected by members in an onboarding prompt.
```
Comment (FedorLap2006) [inline structs.go:1261]:
```suggestion
	// RoleFlagInPrompt indicates a Role to be selectable by members in an onboarding prompt.
```

Comment (Earlopain) [reply structs.go:1261]: I copied the description straight from the discord docs like I usually do. Any reason to deviate here?

File: structs.go [hunk 3]
```diff
@@ -1355,8 +1355,23 @@ type Role struct {
...
 	// The emoji assigned to this role.
 	UnicodeEmoji string `json:"unicode_emoji"`
+
+	// The flags of the role, which describe extra features of a role.
+	// This is a combination of bit masks; the presence of a certain permission can
+	// be checked by performing a bitwise AND between this int and the flag.
+	Flags RoleFlags `json:"flags"`
 }
 
+// RoleFlags is the flags of "role" (see RoleFlags* consts)
```
Comment (FedorLap2006) [inline structs.go:1365]:
```suggestion
// RoleFlags represents the flags of a Role.
```

File: structs.go [hunk 4]
```diff
@@ -1355,8 +1355,23 @@ type Role struct {
...
+// https://discord.com/developers/docs/topics/permissions#role-object-role-flags
+type RoleFlags int
+
+// Block containing known RoleFlags values.
+const (
+	// RoleFlagInPrompt indicates a Role to be selectable by members in an onboarding prompt.
```
Comment (FedorLap2006) [inline structs.go:1371]:
```suggestion
	// RoleFlagInPrompt indicates a Role is selectable by members in an onboarding prompt.
```

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]

**Pull request merged**